libcxx/include/__fwd/string.h

```diff
@@ -45,6 +47,10 @@ class _LIBCPP_TEMPLATE_VIS allocator;
 template <class _CharT, class _Traits = char_traits<_CharT>, class _Allocator = allocator<_CharT> >
 class _LIBCPP_TEMPLATE_VIS basic_string;
 
+// TODO: This could be extended to also allow custom allocators and possibly custom chars and char traits
+template <class _CharT>
```

Comment:
As mentioned in the live review, I think we can also support arbitrary char traits at no additional cost here. With a comment explaining why this is all good.